### PR TITLE
Add variables in inventory generation

### PR DESCRIPTION
**Pull request by 14rcole**
Allow users to define variables in `cfgs` section of the PinFile, which can be used to customize inventory output

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@14rcole : it would be great if you break it down into multiple commits . 
For example:
It better if we have multiple commits for multiple types of files such as 
- tests
- examples 
- code 
    - provider
    - schema

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by herlo** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by herlo** [comment]
@14rcole per our conversation on bjns today, I think we can take the cfgs and config items being passed around and integrate them into the context object like we do with other things. This will provide us with a way forward for other methods that may use this functionality.

**Comment by herlo** [comment]
@14rcole bump

**Comment by herlo** [comment]
@14rcole could you rebase against develop?

**Comment by herlo** [comment]
@samvarankashyap bump

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
@p3ck @samvarankashyap ptal.  I'd like to get a more up to date review since the approving review is now a few weeks old

File: linchpin/provision/roles/beaker/tasks/provision_resource_group.yml [hunk 1]
```diff
@@ -57,9 +57,9 @@
     set_fact:
       bkr_host_values: []
 
-  - name: "combine bkr_hosts and bkr_names"
-    set_fact:
-      bkr_host_values: "{{ bkr_hosts | combine_hosts_names(bkr_names) }}"
+#  - name: "combine bkr_hosts and bkr_names"
```
Comment (samvarankashyap) [inline linchpin/provision/roles/beaker/tasks/provision_resource_group.yml]: @14rcole so we dont need this anymore ??

File: linchpin/InventoryFilters/InventoryFilter.py [hunk 2]
```diff
@@ -109,3 +109,45 @@ def add_common_vars(self, host_groups, layout):
...
+                return (var, val)
+        return ''
+
+    def set_config_values(self, host_data, instance, cfgs={}):
+        """
+        finds the key associated with each config variable and get the
+        value
+        """
+        if cfgs is None:
+            print "cfgs is None"
```
Comment (samvarankashyap) [inline linchpin/InventoryFilters/InventoryFilter.py]: Print statements are not python3 compatible
Please use
print() instead
@14rcole

Comment (14rcole) [reply linchpin/InventoryFilters/InventoryFilter.py]: That was a debugging statement anyway.  Fixed now

**Comment by samvarankashyap** [comment]
@14rcole : build fails

**Comment by 14rcole** [comment]
@samvarankashyap good to merge?

**Comment by 14rcole** [comment]
@p3ck Is the failure on upshift? It looks like the centos tests passed

**Comment by p3ck** [comment]
@14rcole yes - the link is just above your comment about merging.

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**